### PR TITLE
Create basic batch ingest sample

### PR DESCRIPTION
**Pull request by chrisjs**
* Read from a file, apply processing, write to database
* Hook in Spring Cloud Task
* Ensure job can be registered and executed through data flow
* Create README documenting build steps and how to register / launch the job as a task via Data Flow shell

### CODE REVIEW

**Comment by chrisjs** [comment]
demo code is the same as in https://github.com/spring-cloud/spring-cloud-dataflow-samples/pull/36. this new PR was created to modify the original demo PR to account for updated project structure / asciidoc

File: src/main/asciidoc/tasks/file-ingest/main.adoc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+* A Running Data Flow Shell
+include::{docs_dir}/shell.adoc[]
+
+* Review the demo source code to understand how to create a data processing batch job application.
+
+==== Building and Running the Demo
+
+. Build the demo JAR
++
```
Comment (dturanski) [inline src/main/asciidoc/tasks/file-ingest/main.adoc]: Should mention the Demo project is in `batch/file-ingest`

Comment (dturanski) [inline src/main/asciidoc/tasks/file-ingest/main.adoc]: Formatting looks good.  Can you add more content about what the app actually does? Walk through the code a bit more. See `streaming custom spring cloud stream processor`.

Comment (chrisjs) [reply src/main/asciidoc/tasks/file-ingest/main.adoc]: sure - added an overview section that describes the application, and provides a brief overview of the key classes and what they do

Comment (chrisjs) [reply src/main/asciidoc/tasks/file-ingest/main.adoc]: done - added in overview section

**Comment by chrisjs** [comment]
@dturanski is this good to be merged?

**Comment by dturanski** [comment]
LGTM; Merged

**Pull request closed without merging**